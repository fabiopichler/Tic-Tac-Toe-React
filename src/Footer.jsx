import React from 'react';
import clsx from 'clsx';

import { Alert, Button } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSun, faMoon } from '@fortawesome/free-solid-svg-icons';

const Footer = ({
    game,
    onNewGame,
}) => (
    <footer className="footer mb-3">
        {game.winner === 0 ? (
            <div className="footer-players">
                <div className={clsx({ selected: game.player === 1 })}>
                    <FontAwesomeIcon
                        icon={faSun}
                        className="mr-2"
                    />

                    Player 1
                </div>

                <div className={clsx({ selected: game.player === 2 })}>
                    <FontAwesomeIcon
                        icon={faMoon}
                        className="mr-2"
                    />
                    
                    Player 2
                </div>
            </div>
        ) : (
            <>
                {game.winner > 0 ? (
                    <Alert
                        variant="success"
                        className="text-center mt-3"
                    >
                        <Alert.Heading>Vitória</Alert.Heading>

                        <div>Player {game.winner} ganhou o jogo</div>
                    </Alert>
                ) : (
                    <Alert
                        variant="danger"
                        className="text-center mt-3"
                    >
                        <Alert.Heading>Fim do jogo</Alert.Heading>
                        
                        <div>Deu empate</div>
                    </Alert>
                )}
            </>
        )}

        <div className="text-center mb-3">
            <Button onClick={onNewGame}>Reiniciar jogo</Button>
        </div>

        <div className="text-center text-secondary small">
            &copy; 2020 Fábio Pichler
        </div>
    </footer>
);

export default Footer;
