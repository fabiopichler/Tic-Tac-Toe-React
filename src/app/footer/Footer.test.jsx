import React from 'react';
import renderer from 'react-test-renderer';

import Footer from './Footer';

test('Footer: renders correctly', () => {
    const component = renderer.create(
        <Footer
            game={{
                player: 1,
                round: 0,
                winner: 0,
                board: [
                    [0, 0, 0],
                    [0, 0, 0],
                    [0, 0, 0],
                ],
            }}
            onNewGame={() => { }}
        />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
