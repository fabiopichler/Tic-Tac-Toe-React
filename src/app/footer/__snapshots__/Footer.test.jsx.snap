// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer: renders correctly 1`] = `
<footer
  className="footer my-4"
>
  <div
    className="text-center mb-3"
  >
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reiniciar jogo
    </button>
  </div>
  <div
    className="text-center text-secondary small"
  >
    © 2020 Fábio Pichler
  </div>
</footer>
`;
